### PR TITLE
Column lineage point in time java client

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
### Problem

Enable getting column lineage point-in-time in Java client of Marquez.

Closes: #2114

### Solution

Implement point-in-time capability within Java client. 

> **Note:** All database schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

File: clients/java/src/main/java/marquez/client/MarquezClient.java [hunk 1]
```diff
@@ -115,50 +116,12 @@ public enum SortDirection {
     @Getter public final String value;
   }
 
-  public Lineage getColumnLineageByDataset(
```
Comment (wslulciuc) [inline clients/java/src/main/java/marquez/client/MarquezClient.java:118]: I would make sure to note the removal of these methods in our changelog.

**Review by wslulciuc — CHANGES_REQUESTED** [review]
I would make sure to note the removal of methods `Client.getColumnLineageByDataset()`, etc (now just `Client.getColumnLineage(NodeId)`) in our changelog.

File: clients/java/src/test/java/marquez/client/MarquezClientTest.java [hunk 2]
```diff
@@ -955,120 +983,21 @@ public void testCreateTag() throws Exception {
...
-    MarquezClient.Lineage lineage = new MarquezClient.Lineage(ImmutableSet.of(node));
-    String lineageJson = lineage.toJson();
-    when(http.get(
-            buildUrlFor(
-                "/column-lineage?nodeId=job%3Anamespace%3Ajob&depth=20&withDownstream=false")))
-        .thenReturn(lineageJson);
-
-    Node retrievedNode =
-        client.getColumnLineageByJob("namespace", "job").getGraph().stream().findAny().get();
-    assertThat(retrievedNode).isEqualTo(node);
```
Comment (wslulciuc) [inline clients/java/src/test/java/marquez/client/MarquezClientTest.java:1071]: Nice bit of test code simplicity ❤️

Comment (pawel-big-lebowski) [reply clients/java/src/test/java/marquez/client/MarquezClientTest.java:1071]: Thanks. I've added note on that to `changelog`.

**Review by wslulciuc — APPROVED** [review]
LGTM 👍

**Pull request merged**